api/envoy/config/core/v3/protocol.proto

```diff
@@ -123,6 +123,23 @@ message UpstreamHttpProtocolOptions {
       [(validate.rules).string = {well_known_regex: HTTP_HEADER_NAME ignore_empty: true}];
 }
 
+// Allows pre-populating the cache with HTTP/3 alternate protocols entries.
+// This will cause Envoy to attempt HTTP/3 to those upstreams, even if the upstreams have not
+// advertised HTTP/3 support.
+// As with regular cached entries, if the origin response would result in clearing an existing
+// alternate protocol cache entry, pre-populated entries will also be cleared.
+// Adding a cache entry with hostname=foo.com port=123 is the equivalent of getting
+// response headers
+// alt-svc: h3=:"123"; ma=86400" in a response to a request to foo.com:123
+message AlternateProtocolsCacheEntry {
```

Comment:
Presumably if we supported both "h3" and "h3-2" we would add some sort of alpn field to this message?

Reply:
Yeah.  Ditto alternate ports and hostnames.  When/if we add it we can just default to h3